File: Runtime/Plugins/unity.androidlib/src/main/java/com/affise/attribution/unity/common/MethodError.java
package com.affise.attribution.unity.common;

public class MethodError extends Exception {
    private String errorCode = "MethodError";
    private String errorMessage = "";
    private Object errorDetails = null;

    public MethodError(String message) {
        super(message);
        this.errorMessage = message;
    }

    public MethodError(String errorCode, String errorMessage, Object errorDetails) {
        super(errorCode);
        this.errorCode = errorCode;
        this.errorMessage = errorMessage;
        this.errorDetails = errorDetails;
    }

    public String getErrorCode() {
        return errorCode;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public Object getErrorDetails() {
        return errorDetails;
    }
}

File: Runtime/Plugins/iOS/Source/common/MethodError.swift
internal class MethodError {
    private var errorCode: String = "MethodError"
    private var errorMessage: String? = nil
    private var errorDetails: Any? = nil

    init(message: String) {
        errorMessage = message
    }

    init(errorCode: String, errorMessage: String?, errorDetails: Any?) {
        self.errorCode = errorCode
        self.errorMessage = errorMessage
        self.errorDetails = errorDetails
    }

    func  getErrorCode() -> String {
        return errorCode
    }

    func getErrorMessage() -> String? {
        return errorMessage
    }

    func getErrorDetails() -> Any? {
        return errorDetails
    }
}
